### PR TITLE
Tick 0005 create new categories and products

### PR DESCRIPTION
**Pull request by DanMedl**
Tick 0005 create new categories and products

### CODE REVIEW

File: com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+import com.atb.daos.ProductDao;
+import com.atb.daos.initializers.impl.CategoryInitializer;
 import com.atb.dtos.Category;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+
+import java.util.*;
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java:9]: imports with * usually forbidden. Please, change your IDE settings

File: com.atb/src/main/java/com/atb/Application.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.atb.daos.CategoryDao;
+import com.atb.daos.ProductDao;
+import com.atb.daos.impl.DefaultCategoryDao;
+import com.atb.daos.impl.DefaultProductDao;
+import com.atb.dtos.Category;
+
+import java.util.List;
+
+public class Application {
+    public static void main(String[] args) {
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/Application.java:12]: Please, format your code

File: com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
-    private Map<String, Category> categoryStorage = new HashMap<>();
+    private Map<String, Category> categoryStorage = new LinkedHashMap<>();
+
+    public DefaultCategoryDao(ProductDao productDao) {
+        CategoryInitializer initializer = new CategoryInitializer();
+        initializer.setProductDao(productDao);
+        initializer.setCategoryDao(this);
+        initializer.initialize();
+
+
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java:21]: remove empty lines

File: com.atb/src/main/java/com/atb/daos/initializers/impl/ProductInitializer.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        fifthProduct.setCode("P0005");
+        fifthProduct.setDescription("Onion");
+        fifthProduct.setPrice(7.0);
+
+        productDao.addProduct(firstProduct);
+        productDao.addProduct(secondProduct);
+        productDao.addProduct(thirdProduct);
+        productDao.addProduct(fourthProduct);
+        productDao.addProduct(fifthProduct);
+
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/initializers/impl/ProductInitializer.java:44]: Please, format your code and remove empty lines

File: com.atb/src/main/java/com/atb/dtos/Category.java [hunk 5]
```diff
@@ -53,4 +53,9 @@ public List<Product> getProducts() {
     public void setProducts(List<Product> products) {
         this.products = products;
     }
+
+    @Override
+    public String toString() {
+        return "Category{" + "code='" + code + '\'' + ", description='" + description + '\'' + ", products=" + products + '}';
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/dtos/Category.java:59]: Wow, good job :)

File: com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java (hunk 1 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java:9]: Done

File: com.atb/src/main/java/com/atb/Application.java (hunk 2 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/Application.java:12]: done

File: com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java (hunk 3 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/impl/DefaultCategoryDao.java:21]: Done

File: com.atb/src/main/java/com/atb/daos/initializers/impl/CategoryInitializer.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        firstCategory.setProducts(Arrays.asList(product));
+
+        Category secondCategory = new Category();
+        secondCategory.setCode("C0002");
+        secondCategory.setDescription("Sort of Cucumber's");
+        Product secondCategoryProduct = productDao.getProductByCode("P0002");
+        secondCategory.setProducts(Arrays.asList(secondCategoryProduct));
+        secondCategory.setParentCategory(firstCategory);
+
+
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/initializers/impl/CategoryInitializer.java:34]: Done

File: com.atb/src/main/java/com/atb/daos/initializers/impl/ProductInitializer.java (hunk 4 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/initializers/impl/ProductInitializer.java:44]: Done

File: com.atb/src/main/java/com/atb/dtos/Category.java (hunk 5 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/dtos/Category.java:59]: Thank you :)

**Pull request merged**